src/Cli/Microsoft.TemplateEngine.Cli/Commands/NewCommand.Legacy.cs

```diff
@@ -79,7 +79,12 @@ internal void ValidateArgumentsAreNotUsed(CommandResult commandResult)
 
         internal void ValidateOptionUsage(CommandResult commandResult, Option option)
         {
-            OptionResult? optionResult = commandResult.Parent?.Children.FirstOrDefault(symbol => symbol.Symbol == option) as OptionResult;
+            if (commandResult.Parent is not CommandResult parentResult)
```

Comment:
A parent of `CommandResult` can be only another `CommandResult`. In case it's null it means it's `RootCommand`.